src/util/tools.cc

```diff
@@ -1283,6 +1283,19 @@ std::string getAVIFourCC(const fs::path& avi_filename)
 }
 #endif
 
+std::string getHostName(const struct sockaddr* addr)
+{
+    char hoststr[NI_MAXHOST];
+    char portstr[NI_MAXSERV];
+    int len = addr->sa_family == AF_INET6 ? sizeof(struct sockaddr_in6) : sizeof(struct sockaddr_in);
+    int ret = getnameinfo(addr, len, hoststr, sizeof(hoststr), portstr, sizeof(portstr), NI_NOFQDN);
```

Comment:
@KarlStraussberger 
This may look like a nice feature but `getnameinfo` is blocking and in case of DNS service unavailable this can freeze the thread for a few minutes.

I would be really cautious about using such innocent-looking non-async APIs all over the place...

Reply:
Oh, and in case of failure you just return garbage :(